evm-contracts/src/v0.6/dev/FluxAggregator.sol

```diff
@@ -609,71 +615,76 @@ contract FluxAggregator is AggregatorInterface, Owned {
    * Private
    */
 
-  function initializeNewRound(uint32 _id)
+  function initializeNewRound(uint32 _roundId)
     private
   {
-    updateTimedOutRoundInfo(_id.sub(1));
+    updateTimedOutRoundInfo(_roundId.sub(1));
 
-    reportingRoundId = _id;
-    rounds[_id].details.maxAnswers = maxAnswerCount;
-    rounds[_id].details.minAnswers = minAnswerCount;
-    rounds[_id].details.paymentAmount = paymentAmount;
-    rounds[_id].details.timeout = timeout;
-    rounds[_id].startedAt = uint64(block.timestamp);
+    reportingRoundId = _roundId;
+    rounds[_roundId].details.maxAnswers = maxAnswerCount;
+    rounds[_roundId].details.minAnswers = minAnswerCount;
+    rounds[_roundId].details.paymentAmount = paymentAmount;
+    rounds[_roundId].details.timeout = timeout;
+    rounds[_roundId].startedAt = uint64(block.timestamp);
 
-    emit NewRound(_id, msg.sender, rounds[_id].startedAt);
+    emit NewRound(_roundId, msg.sender, rounds[_roundId].startedAt);
   }
 
-  function oracleInitializeNewRound(uint32 _id)
+  function oracleInitializeNewRound(uint32 _roundId)
     private
-    ifNewRound(_id)
-    ifOracleDelayed(_id)
   {
-    initializeNewRound(_id);
+    if (!newRound(_roundId)) return;
+    uint256 lastStarted = oracles[msg.sender].lastStartedRound; // cache storage reads
+    if (_roundId <= lastStarted + restartDelay && lastStarted != 0) return;
+
+    initializeNewRound(_roundId);
 
-    oracles[msg.sender].lastStartedRound = _id;
+    oracles[msg.sender].lastStartedRound = _roundId;
   }
 
-  function requesterInitializeNewRound(uint32 _id)
+  function requesterInitializeNewRound(uint32 _roundId)
     private
-    ifNewRound(_id)
-    onlyDelayedRequesters(_id)
   {
-    initializeNewRound(_id);
+    if (!newRound(_roundId)) return;
+    uint256 lastStarted = requesters[msg.sender].lastStartedRound; // cache storage reads
+    require(_roundId > lastStarted + requesters[msg.sender].delay || lastStarted == 0, "must delay requests");
 
-    requesters[msg.sender].lastStartedRound = _id;
+    initializeNewRound(_roundId);
+
+    requesters[msg.sender].lastStartedRound = _roundId;
   }
 
-  function updateTimedOutRoundInfo(uint32 _id)
+  function updateTimedOutRoundInfo(uint32 _roundId)
     private
-    ifTimedOut(_id)
-    onlyWithPreviousAnswer(_id)
```

Comment:
Just noting that the require statement from `onlyWithPreviousAnswer` is no longer included in this function.

Reply:
Since we started setting the previous answer in the constructor, this is always true and so not a helpful check.

Reply:
The modifier actually checked that updated at was set for the previous round, and used that as a sign of being answered. Since we started setting the updated at in the constructor, this is always true and so not a helpful check.